packages/@lwc/template-compiler/src/parser/experimental.ts

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import * as parse5 from 'parse5-with-errors';
+
+import {
+    CompilerDiagnostic,
+    generateCompilerDiagnostic,
+    LWCErrorInfo,
+    ParserDiagnostics,
+} from '@lwc/errors';
+
+import { parseHTML, traverseHTML, treeAdapter } from './html';
+
+function filter(node: parse5.AST.Default.Element) {
+    delete node.parentNode;
+    delete node.__location;
+
+    let nodeToFilter = node;
+
+    const content = treeAdapter.getTemplateContent(node);
+    if (content) {
+        nodeToFilter = content as parse5.AST.Default.Element;
+    }
+
+    filterChildNodes(nodeToFilter);
+}
+
+function filterChildNodes(node: parse5.AST.Default.Element) {
+    node.childNodes = node.childNodes.filter(childNode => {
+        return !treeAdapter.isTextNode(childNode) && !treeAdapter.isCommentNode(childNode);
+    });
+}
+
+export function experimentalParse(source: string) {
+    const warnings: CompilerDiagnostic[] = [];
+    const { fragment, errors: parsingErrors } = parseHTML(source);
+
+    if (parsingErrors.length !== 0) {
+        return { warnings: parsingErrors };
+    }
+
+    const templateRoot = getTemplateRoot(fragment);
+    if (!templateRoot) {
+        return { warnings };
```

Comment:
shouldn't we produce a warning here? The warnings property is initialized to [] and is never populated, which means we will be returning an empty list.

Reply:
The array is populated during the call to `getTemplateRoot`.